packages/transit-vehicle-overlay/src/utils/state.js

```diff
@@ -0,0 +1,160 @@
+import { useState, useRef, useEffect, useCallback } from "react";
+import { checkRefreshInteval } from "./fetch";
+import * as data from "./data";
+
+/**
+ * use this hook when you want your layer to re-paint after otp-ui map zoom events fire
+ * this component will capture and save zoom state from otp-ui's base-map component to state
+ * :return: zoom level (number from leaflet) and the callback for base-map's onViewportChanged
+ */
+export function zoomState(initialZoom = 14) {
+  const [mapZoom, setMapZoom] = useState(initialZoom);
+
+  const onViewportChanged = ({ zoom }) => {
+    // console.info(`zoom level ${zoom}`);
+    setMapZoom(zoom);
+  };
+
+  return [mapZoom, onViewportChanged];
+}
+
+/**
+ * use this hook when you want your layer to re-paint after otp-ui map zoom & recenter events fire
+ * this component will capture and save zoom and center from otp-ui's base-map viewport changes
+ * :return: zoom level, center [x,y] and the callback for base-map's onViewportChanged
+ */
```

Comment:
Might be useful to distinguish this jsdoc a little more explicitly from the above hook. Also, injecting some punctuation and caps might help legibility. Finally, the `:return:` syntax seems non-standard. Should this be `@returns`?